lightning-block-sync/src/http.rs

```diff
@@ -339,17 +334,34 @@ impl HttpClient {
 							reader.read_exact(&mut content[chunk_offset..]).await?;
 							content.resize(chunk_offset + chunk_size, 0);
 						}
-						Ok(content)
+						content
 					}
 					#[cfg(not(feature = "tokio"))]
 					{
 						let mut decoder = chunked_transfer::Decoder::new(reader);
 						decoder.read_to_end(&mut content)?;
-						Ok(content)
+						content
 					}
 				}
 			},
+		};
+
+		if !status.is_ok() {
+			// TODO: Handle 3xx redirection responses.
+			let mut all_ascii = true;
+			for byte in contents.iter() {
+				if !byte.is_ascii() { all_ascii = false; }
+			}
```

Comment:
Would `is_ascii` on a slice work?

https://doc.rust-lang.org/std/primitive.slice.html#method.is_ascii

Reply:
Perf, thank you!